platform/android/export/export.cpp

```diff
@@ -52,7 +52,9 @@ void register_android_exporter() {
 #ifndef ANDROID_ENABLED
 	EDITOR_DEF_BASIC("export/android/java_sdk_path", OS::get_singleton()->get_environment("JAVA_HOME"));
 	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::STRING, "export/android/java_sdk_path", PROPERTY_HINT_GLOBAL_DIR));
-	EDITOR_DEF_BASIC("export/android/android_sdk_path", OS::get_singleton()->get_environment("ANDROID_HOME"));
+
+	EDITOR_DEF_BASIC("export/android/android_sdk_path", OS::get_singleton()->has_environment("ANDROID_HOME") ? OS::get_singleton()->get_environment("ANDROID_HOME") : OS::get_singleton()->get_default_android_sdk_path());
```

Comment:
I wonder why the Android installers don't define `ANDROID_HOME` to the user environment variables after installing Android Studio. If they consistently did this, then the step wouldn't be needed.

Reply:
I thing it's due to Linux's mess in environment variables. While in windows they are stored compactly and conveniently in one place in registry, in Linux there plenty of places for both user and global variables. And what makes it even worse is the fact that this set of places depends on DE and shell you currently use (and maybe distro, if we're taking about global vars). Yes, we have '/etc/environment', but you need sudo to write to it. So there's very few examples of non system apps which automatically edit variables

Reply:
> Yes, we have '/etc/environment', but you need sudo to write to it.

On macOS, it's not writable at all, there's no way to set global environment variables. `ANDROID_HOME` can be set for the shell, but Godot usually is not started from the shell.